app/models/location_report.rb

```diff
@@ -0,0 +1,208 @@
+# This file is part of SEQUENCESCAPE; it is distributed under the terms of
+# GNU General Public License version 1 or later;
+# Please refer to the LICENSE and README files for information on licensing and
+# authorship of this file.
+# Copyright (C) 2018 Genome Research Ltd.
+
+class LocationReport < ApplicationRecord
+  extend DbFile::Uploader
+
+  attr_accessor :barcodes_list
+
+  def barcodes_text
+    barcodes_list.join(' ') unless @barcodes_list.nil?
+  end
+
+  def barcodes_text=(value)
+    self.barcodes_list = if value.nil?
+                           nil
+                         else
+                           value.strip.split(/[\s]+/)
+                         end
+  end
+
+  self.per_page = 10
+
+  scope :for_user, ->(user) { where(user_id: user.id) }
+
+  belongs_to :study
+  belongs_to :user
+
+  has_uploaded :report, serialization_column: 'report_filename'
+
+  def headers
+    %w[Barcode HumanBarcode Type Created Location Service Study Owner]
+  end
+
+  def generate!
+    ActiveRecord::Base.transaction do
+      csv_options = { row_sep: "\r\n", force_quotes: true }
+      filename    = "location_report_#{name}.csv"
+      Tempfile.open(filename) do |tempfile|
+        generate_report_rows do |fields|
+          tempfile.puts(CSV.generate_line(fields, csv_options))
+        end
+        tempfile.open # Reopen the temporary file
+        update_attributes!(report: tempfile)
+      end
+    end
+  end
+
+  serialize :plate_purpose_ids, Array
+
+  before_validation :generate_name, on: %i[create build]
+
+  after_create :schedule_report
+
+  validates :report_type, presence: true
+
+  validates :study_id, numericality: true, if: :report_type_selection?, allow_nil: true
+  validates :start_date, presence: { if: :report_type_selection? }
+  validates :end_date, presence: { if: :report_type_selection?, date: { after_or_equal_to: :start_date } }
+
+  validates :barcodes_text, presence: { if: :report_type_barcodes? }
+  validate :barcodes_are_recognised, if: :report_type_barcodes?
+  validate :plates_are_found, if: :report_type_selection?
+
+  def report_type_selection?
+    report_type == 'selection'
+  end
+
+  def report_type_barcodes?
+    report_type == 'barcodes'
+  end
+
+  def barcodes_are_recognised
+    if barcodes_list.blank?
+      errors.add(:barcodes_text, 'Please enter some barcodes in the text area')
+      return
```

Comment:
http://guides.rubyonrails.org/i18n.html
But this has already been rested above, so probably don't need to add the error again. YOu might even be able to tweak the if: above to only run if the text is filled.